scripts/update_deps.py

```diff
@@ -151,53 +185,96 @@ def find_latest_libtpu_nightly() -> Optional[tuple[str, str, str]]:
       _PLATFORM + r'\.whl</a>')
 
 
+def fetch_pep503_page(url: str) -> list[tuple[str, str]]:
+  """Fetches and parses a PEP 503 index page.
+
+  Args:
+    url: The URL of the PEP 503 index page.
+
+  Returns:
+    A list of (href, text) tuples for all links on the page.
+  """
+  try:
+    with urllib.request.urlopen(url) as response:
+      html = response.read().decode('utf-8')
+
+    parser = PEP503Parser()
+    parser.feed(html)
+    return parser.links
+  except Exception as e:
+    logger.error(f'Failed to fetch {url}: {e}')
+    return []
+
+
 def find_latest_jax_nightly() -> Optional[tuple[str, str, str]]:
-  """Finds the latest JAX nightly build.
-  
+  """Finds the latest JAX nightly build using the new package index.
+
   Returns:
     A tuple of the jax version, jaxlib version, and date of the latest JAX nightly build,
     or None if no build is found.
   """
 
-  # Read the nightly jax build page.
-  clean_tmp_dir()
-  os.system('curl -s {} > {}/jax_builds.html'.format(_JAX_BUILDS_URL, _TMP_DIR))
-  with open(f'{_TMP_DIR}/jax_builds.html', 'r') as f:
-    html_lines = f.readlines()
+  # Fetch JAX project page
+  jax_links = fetch_pep503_page(_JAX_PROJECT_URL)
+  if not jax_links:
+    logger.error(f'Could not fetch JAX packages from {_JAX_PROJECT_URL}')
+    return None
 
-  # Find lines like
-  # <a href=...>jax/jax-0.6.1.dev20250428-py3-none-any.whl</a>
-  jax_build = find_latest_nightly(
-      html_lines, r'.*<a href=.*?>jax/jax-(.*?)\.dev(\d{8})-(.*)\.whl</a>')
-  if not jax_build:
-    logger.error(
-        f'Could not find latest jax nightly build in {_JAX_BUILDS_URL}.')
+  # Fetch jaxlib project page
+  jaxlib_links = fetch_pep503_page(_JAXLIB_PROJECT_URL)
+  if not jaxlib_links:
+    logger.error(f'Could not fetch jaxlib packages from {_JAXLIB_PROJECT_URL}')
     return None
 
-  # Find lines like
-  # <a href=...>nocuda/jaxlib-0.6.1.dev20250428-....whl</a>
-  jaxlib_build = find_latest_nightly(
-      html_lines,
-      r'.*<a href=.*?>nocuda/jaxlib-(.*?)\.dev(\d{8})-(.*)\.whl</a>')
-  if not jaxlib_build:
-    logger.error(
-        f'Could not find latest jaxlib nightly build in {_JAX_BUILDS_URL}.')
+  # Parse JAX nightly versions
+  # Looking for patterns like: jax-0.6.1.dev20250428-py3-none-any.whl
+  jax_pattern = re.compile(r'jax-([\d.]+)\.dev(\d{8})-py3-none-any\.whl')
+  latest_jax_version = ''
+  latest_jax_date = ''
+
+  for href, text in jax_links:
+    filename = text if text else href.split('/')[-1].split('#')[0]
+    m = jax_pattern.match(filename)
+    if m:
+      version, date = m.groups()
+      if date > latest_jax_date:
+        latest_jax_version = version
+        latest_jax_date = date
+
+  if not latest_jax_version:
+    logger.error('Could not find any JAX nightly builds')
     return None
 
-  jax_version, jax_date, _ = jax_build
-  jaxlib_version, jaxlib_date, _ = jaxlib_build
-  if jax_date != jaxlib_date:
+  # Parse jaxlib nightly versions
```

Comment:
Consider refactor to share logic for parsing jax and jaxlib. This can be a TODO.